modules/createHistory.js

```diff
@@ -114,6 +114,16 @@ function createHistory(options={}) {
         return // Transition was interrupted.
 
       if (ok) {
+        // treat PUSH to current path like REPLACE to be consistent with browsers
+        if (nextLocation.action === PUSH) {
+          let { pathname, search } = getCurrentLocation()
+          let currentPath = pathname + search
+          let path = nextLocation.pathname + nextLocation.search
```

Comment:
I think `createPath` is the correct check here. It doesn't account for `basename` (and that part is a bit messy honestly in `createBrowserHistory`), but `basename` isn't going to change on us anyway, so it doesn't matter. That would also fix https://github.com/rackt/history/pull/43#issuecomment-160782080.